### PR TITLE
ref(new-widget-builder-experience): Add drag-and-drop to the groupByField - [DD-721]

### PR DESCRIPTION
**Pull request by priscilawebdev**
**What this PR does**

- Add drag and drop feature to the groupBy field in the widget builder
- Filter grouoBy filter options to not show on the list already selected values
- Unified the `releaseGroupBySelector` into the `groupBySelector` component 
- Add a test to the widget builder file 

**Preview:**

https://user-images.githubusercontent.com/29228205/163995583-00b1f139-5394-4076-b0d3-a98d69516d05.mov

### CODE REVIEW

File: tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx [hunk 1]
```diff
@@ -2104,6 +2104,31 @@ describe('WidgetBuilder', function () {
...
+      expect(screen.queryByLabelText('Drag to reorder')).not.toBeInTheDocument();
+
+      userEvent.click(screen.getByText('Add Group'));
+
+      expect(screen.getAllByLabelText('Remove group')).toHaveLength(2);
+      expect(screen.getAllByLabelText('Drag to reorder')).toHaveLength(2);
+    });
+
+    it.todo(
+      'Since simulate drag and drop with RTL is not recommended because of browser layout, remember to create acceptance test for this'
```
Comment (priscilawebdev) [inline tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx:2130]: for more context https://github.com/testing-library/user-event/issues/440#issuecomment-685010755

**Comment by edwardgou-sentry** [comment]
I noticed some fields in the groupby have a tags badge when selected
image
image

**Comment by priscilawebdev** [comment]
good catch @edwardgou-sentry !

**Review by edwardgou-sentry — APPROVED** [review]
I think the code looks good to me, and the group by badges also looks fixed to me now.

Maybe a silly question, but just for my understanding, does the order of the group by have any impact on the actual query and results?

**Comment by priscilawebdev** [comment]
@narsaynorath can maybe help us here

**Comment by edwardgou-sentry** [comment]
@narsaynorath sounds good, thanks!

**Pull request merged**